code/controllers/subsystem/processing/orbits.dm

```diff
@@ -151,3 +160,89 @@ PROCESSING_SUBSYSTEM_DEF(orbits)
 	for(var/obj/machinery/computer/objective/computer as() in GLOB.objective_computers)
 		for(var/M in computer.viewing_mobs)
 			computer.update_static_data(M)
+
+//Saves custom shuttles
+/datum/controller/subsystem/processing/orbits/proc/save_custom_shuttles()
+
+	if(!CONFIG_GET(flag/save_shuttle_ruins))
+		message_admins("Saving shuttles skipped, it is disabled in the config!")
+		return
+
+	//Calculate saved shuttle filesize
+	var/shuttle_filepath = CONFIG_GET(string/shuttle_ruin_filepath)
+
+	//Save shuttle
+	message_admins("SSORBITS: Saving custom shuttle ruins...")
+
+	var/list/old_files = list()
+	var/current_size = 0
+
+	//Find the current filesize saved
+	if(fexists(shuttle_filepath))
+		var/list/files = flist(shuttle_filepath)
+		//Calculate length of files
+		for(var/f in files)
+			var/fullF = "[shuttle_filepath][f]"
+			var/filelength = length(file(fullF))
+			old_files += fullF
+			current_size += filelength
+		message_admins("SSORBITS: Located [length(files)] saved shuttles, with total filesize of [current_size] bytes!")
+	else
+		message_admins("SSORBITS: No custom shuttle files currently exist on the server!")
+
+	var/custom_shuttle_count = 0
+	var/shuttles_saved = 0
+
+	var/space_required = 0
+
+	var/space_saved = 0
+	var/shuttles_deleted = 0
+
+	//Find shuttles which we want to save
+	for(var/obj/docking_port/mobile/M in SSshuttle.mobile)
+		//Detect real custom shuttles
+		var/turf/T = get_turf(M)
+		if(!istype(T.loc, /area/shuttle/custom))
+			continue
+		//Count custom shuttles
+		custom_shuttle_count ++
+		//Calculate size
+		if(M.width > 35 || M.height > 35)
+			message_admins("Custom shuttle [M] skipped due to being over the size limit of (35x35).")
+			continue
+		//Alright lets get the save data
+		var/save_data = M.get_shuttle_tgm_data()
+		//Calculate filesize (1 byte = 1 char)
+		var/file_size = length(save_data)
+		if(file_size > CONFIG_GET(number/shuttle_filesize_max))
+			message_admins("Custom shuttle [M] skipped due to being over the file size limit of [CONFIG_GET(number/shuttle_filesize_max)] bytes.")
+			continue
+		space_required += file_size
+		//Save the shuttle file
+		var/file_name = "[shuttle_filepath]customshuttle_[GLOB.round_id]_[shuttles_saved].dmm"
+		if(fexists(file_name))
+			fdel(file_name)
+		text2file(save_data, file_name)
+		//Log it
+		shuttles_saved ++
+		message_admins("Custom shuttle [M] successfully saved!")
+		CHECK_TICK
+
+	//Clear old shuttles to free up space
+	var/left_to_clear = max(current_size + space_required - CONFIG_GET(number/shuttle_total_filesize_max), 0)
+	//Luck of the draw
+	shuffle_inplace(old_files)
```

Comment:
Considering the potential size of this list it may be better to just do `rand()` a few times.

Reply:
oh yea probably 1000x better to do old_files[rand(1, length(old_files))]